Automattic/wc-calypso-bridge
# Free trial - payment task replacements

#### Changes

This PR replaces WooCommerce Core's payment task in free trial mode.


2. Copied over [PaymentGatewaySuggestions](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-admin/client/tasks/fills/PaymentGatewaySuggestions) component in https://github.com/Automattic/wc-calypso-bridge/pull/919/commits/9ca4f8a0ca3d752d57329f0d2c319e95ba65dbfe (code changes) and https://github.com/Automattic/wc-calypso-bridge/pull/919/commits/9699044f84e62be5f5ffcd61d5031bedae55089f (npm package changes)
3. Copied over WCPayBannerBody and WCPayBannerText to customize the copies in https://github.com/Automattic/wc-calypso-bridge/pull/919/commits/7d86b13ce489fca4c98594e0b614083c15095333
4. Added trial plan notice in https://github.com/Automattic/wc-calypso-bridge/pull/919/files#diff-dfa9476bc6dd5969d23b283b6728354b0cc05430621cbf8b9834c724bbacc87dR277 (Do we have a component for the notice already? I think I've seen one in Gutenberg, but I wasn't able to locate it) 

6. Tasklist API replacement logic is in https://github.com/Automattic/wc-calypso-bridge/pull/919/files#diff-ffce9353e36a45176f411a50eb1c4dd3bfa91ec8cf2eec0ec2b5e8f8dc7e19b4R41

#### Test instructions



1. Checkout trunk branch in your local WC env
2. Make sure to build assets.
3. Clone this branch in your wc-calypso-bridge
4. Make sure to copy over the latest changes in [woocmmerce-calypso-bridge-helper.php](https://gist.github.com/moon0326/cac46c70a2cee81b61faef517fef7178) and the plugin is activated.
7. Change [this line](https://gist.github.com/moon0326/cac46c70a2cee81b61faef517fef7178#file-a-woocommerce-calypso-bridge-helper-php-L33) to false in your local env. This enables free trial in your local env.
8. Run `npm run start` to build frontend changes.
9. Navigate to `WooCommerce -> Home`
10. Complete store details task with a country that is not wcpay eligible
11. Click payments task.
12. Confirm there is a notice `During the trial period you can only make test payments. To process real transactions, upgrade now.` and links to `https://wordpress.com/plans/<site_slug>` such as `https://wordpress.com/plans/localhost:8888`
13. Confirm there is a description: "To get ready to accept online payments”  blew "Choose a payment provider".

![Screenshot 2023-02-16 at 14 17 01](https://user-images.githubusercontent.com/4344253/219285601-31f90b8b-2048-484c-ba74-551109cf2570.png)


14. Change your store country to US
15. Navigate to `WooCommerce -> Home > payment task`
16. Confirm there is a notice `During the trial period you can only make test payments. To process real transactions, upgrade now.` 
17. Confirm heading text is “Get ready to be paid and manage your business”.

![Screenshot 2023-02-16 at 14 12 32](https://user-images.githubusercontent.com/4344253/219285583-b719ef51-c8fa-4619-b249-53ff513db604.png)
